Dependency registry schema updates

yoshiwarab opened the pull request:
## Description of the change

\<add description here\>

>reminder: Any subgraph deployments should be documented as [Releases](https://github.com/ArtBlocks/artblocks-subgraph/releases) in this repository.

>reminder: Any changes to subgraph schema should be accompanied by corresponding changes to the Art Blocks documentation site, available at https://docs.artblocks.io/ (specifically, see the [Subgraph Entities](https://docs.artblocks.io/creator-docs/art-blocks-api/entities/) and [Subgraph Querying and API Overview](https://docs.artblocks.io/creator-docs/art-blocks-api/queries/) sections).

ryley-o commented on schema.graphql:
```diff
@@ -552,12 +552,43 @@ type Transfer @entity(immutable: true) {
 }
 
+type Dependency @entity {
+  id: ID!
```
nit: can we add a comment specifying the id syntax (for this and the other new ones)?

yoshiwarab replied:
Resolved in https://github.com/ArtBlocks/artblocks-subgraph/pull/168

ryley-o commented on schema.graphql:
```diff
@@ -552,12 +552,43 @@ type Transfer @entity(immutable: true) {
+  preferredRepository: String!
+  additionalRepositoryCount: BigInt!
+  additionalRepositories: [DependencyAdditionalRepository!]! @derivedFrom(field: "dependency")
+  scripts: [DependencyScript!]! @derivedFrom(field: "dependency")
```
do we also want a `script` field here (analogous to `Project.script`)

yoshiwarab replied:
Resolved in https://github.com/ArtBlocks/artblocks-subgraph/pull/159

ryley-o reviewed (approved):
👍

ryley-o commented:
@yoshiwarab - might want to merge this into it's own feature branch?

yoshiwarab commented:
> @yoshiwarab - might want to merge this into it's own feature branch?

Good call, updated!

ryley-o commented:
@yoshiwarab - is this ready to merge? Also wanted to call out that it would be great to keep our docs up to date and include this part of our schema https://docs.artblocks.io/creator-docs/art-blocks-api/entities/

yoshiwarab commented:
> @yoshiwarab - is this ready to merge? Also wanted to call out that it would be great to keep our docs up to date and include this part of our schema https://docs.artblocks.io/creator-docs/art-blocks-api/entities/

@ryley-o Done! https://github.com/ArtBlocks/artblocks-docs/pull/170

Also wrote this little script to take the lines of the schema and spit out a text table (which can then be formatted with the vscode text table extension or emacs table editor.

```
function schemaToTextTable(schemaText) {
    const rows = schemaText.split('\n').map((line) => line.trim()).reduce((acc, line, idx) => {
      console.log(line)
      if (line.startsWith('"')) {
        const fieldObj = {
          description: line.replace(/"/g, ''),
        }
        return acc.concat(fieldObj)
      }

      const fieldName = line.split(':')[0].trim()
      const fieldType = line.split(':')[1].trim()

      let fieldObj
      const prevEntry = acc[acc.length - 1]
      if (prevEntry && !prevEntry.fieldName) {
        fieldObj = prevEntry
        fieldObj.fieldName = fieldName
        fieldObj.fieldType = fieldType
        acc[acc.length - 1] = fieldObj
      } else {
        fieldObj = {
          fieldName,
          fieldType
        }
        acc.concat(fieldObj)
      }
      return acc
    }, []).map((fieldObj) => `| ${fieldObj.fieldName} | ${fieldObj.fieldType} | ${fieldObj.description} |`).join('\n')
    const table = `| Field | Type | Description |\n| ----- | ---- | ----------- |\n${rows}`
    return table
  }```

ryley-o commented:
>Also wrote this little script to take the lines of the schema and spit out a text table (which can then be formatted with the vscode text table extension or emacs table editor.

@yoshiwarab - want to add that script to this repo and put in a command in `package.json`?
